src/EventListener/KernelRequestListener.php

```diff
@@ -0,0 +1,381 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Oveleon\ContaoCookiebar\EventListener;
+
+use Contao\ContentModel;
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Contao\CoreBundle\Security\Authentication\Token\TokenChecker;
+use Contao\Model;
+use Contao\ModuleModel;
+use Contao\PageModel;
+use Contao\StringUtil;
+use Contao\System;
+use Oveleon\ContaoCookiebar\Cookiebar;
+use Oveleon\ContaoCookiebar\Model\CookiebarModel;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class KernelRequestListener
+{
+    private ?string $rootPageBuffer = null;
+    private ?PageModel $objRootPage = null;
+    private ?PageModel $objPage = null;
+    private ?CookiebarModel $cookiebarModel = null;
+    private ?string $globalJavaScript = null;
+
+    public function __construct(
+        private readonly TranslatorInterface $translator,
+        private readonly TokenChecker        $tokenChecker,
+        private readonly ScopeMatcher        $scopeMatcher,
+        private readonly int                 $lifetime,
+        private readonly string              $storageKey,
+        private readonly bool                $considerDnt
+    )
+    {
+    }
+
+    /**
+     * @param RequestEvent $event
+     * @return void
+     */
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        $request = $event->getRequest();
+
+        // Also handles CookieBar not being activated within rootPage
+        if (
+            $this->objRootPage instanceof PageModel ||
+            !$this->scopeMatcher->isFrontendRequest($request) ||
+            !$request->attributes->has('pageModel')
+        )
+        {
+            return;
+        }
+
+        // use first Contao-FrontendRequest to handle the pageModel settings
+        // normally this should be the MainRequest to handle Page
+        // check also pageType because it also could be a rootPage which is redirected then
+        $pageModel = $request->attributes->get('pageModel');
+        if (
+            $pageModel instanceof PageModel &&
+            $pageModel->type === 'regular'
```

Comment:
This can be removed as the cookiebar can be shown on 401,403,404 and custom page types as well.